File: src/main/java/com/upokecenter/text/ICharacterDecoder.java
package com.upokecenter.text;

import java.io.*;
import com.upokecenter.util.*;

    /**
     * Defines a method that can be implemented by classes that convert a stream of
     * bytes to Unicode code points.
     */
public interface ICharacterDecoder {
    /**
     *
     */
  int ReadChar(IByteReader input);
}

File: Encoding/PeterO/Text/ICharacterDecoder.cs
using System;
using System.IO;
using PeterO;

namespace PeterO.Text {
    /// <include file='../../docs.xml'
  /// path='docs/doc[@name="T:PeterO.Text.ICharacterDecoder"]/*'/>
public interface ICharacterDecoder {
    /// <include file='../../docs.xml'
  /// path='docs/doc[@name="M:PeterO.Text.ICharacterDecoder.ReadChar(PeterO.IByteReader)"]/*'/>
  int ReadChar(IByteReader input);
}
}
